src/main/java/catascore/KeyGeneratorImpl.java

```diff
@@ -1,22 +1,26 @@
 package catascore;
 
-public class KeyGeneratorImpl implements KeyGenerator {
+
+public class KeyGeneratorImpl  implements KeyGenerator{
 
     private int first = 0;
     private StringBuffer key = new StringBuffer("AAAAAA");
     private int currentPos = key.length() - 1;
 
-    @Override
-    public String generateKey() {
-        if( this.first != 0){
-            if(this.key.charAt(this.currentPos) == 'Z'){
-                this.key.setCharAt( this.currentPos, 'A');
-                this.currentPos--;
+    public String generateKey() throws Exception{
```

Comment:
As good practice always a minimum statement of what the method does, input and output parameters

Reply:
Ok, thanks